packages/mask/background/tasks/pending-tasks.ts

```diff
@@ -0,0 +1,29 @@
+import { NetworkPluginID, PersistentStorages } from '@masknet/shared-base'
+import { MessageStateType, type ReasonableMessage } from '@masknet/web3-shared-base'
+
+function checkMessages(messages: Array<ReasonableMessage<Request, Response>>) {
+    const pendingTasks = messages
+        .filter((x) => x.state === MessageStateType.NOT_DEPEND)
+        .sort((a, z) => a.createdAt.getTime() - z.createdAt.getTime())
+    const length = pendingTasks.length
+    browser.browserAction.setBadgeBackgroundColor({
+        color: '#d32f2f',
+    })
+    browser.browserAction.setBadgeText({
```

Comment:
please implement this feature via feature detection.

This feature should support both MV3 API `browser.action` and MV2 `browser.browserAction`. You should prefer MV3 API when existing. (Use `if (browser.action)` to detect).

More information:

https://developer.chrome.com/blog/mv3-actions/#manifest-v3-changes